tests/framework/test_time.py

```diff
@@ -408,6 +433,47 @@ def test_multiple_modifiers(SimulationContext, base_config):
         )
 
 
+def test_untracked_simulants(SimulationContext, base_config):
```

Comment:
This addition is the main component of the tracked/untracked issue. Explained in more detail in the description, this checks that we don't include untracked simulants in the event indices and step size calculations, the default behavior with current functionality.